pkg/registry/pod/strategy.go

```diff
@@ -153,6 +154,23 @@ func (podStatusStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object
 	return validation.ValidatePodStatusUpdate(obj.(*api.Pod), old.(*api.Pod))
 }
 
+func CheckPod(r *registry.Store, ctx api.Context, err error, obj runtime.Object) error {
+	if !errors.IsAlreadyExists(err) {
+		return err
+	}
+	pod := obj.(*api.Pod)
+	retObj, getPodErr := r.Get(ctx, pod.Name)
+	if getPodErr != nil {
+		return err
+	}
+	retPod := retObj.(*api.Pod)
+	if api.IsPodTerminated(retPod) {
+		status_err, _ := err.(*errors.StatusError)
+		status_err.ErrStatus.Message += fmt.Sprintf("\n Existant pod is in terminated state! Please check its status and delete it!\n Use `kubectl get pods/%s` \n", pod.Name)
```

Comment:
There are also performance implications here.  This needs to be handled in a place that the server has responded with the object (in the storage code) so that we don't have to fetch the object a second time.  Loading it again is also racy.

Agree with Derek in that the message cannot describe anything about kubectl - the server is client agnostic.  You can use a more generic error (in which case you could handle it in NewAlreadyExists() with something like `... - the object may be in the process of deletion` or you can handle it in the storage layer (where we have access to both the Decoder and the store) to get additional information out of the object.  Since generic.Store doesn't have access to an encoder it would be inappropriate to decode the error response here.